src/dumpulator/dumpulator.py

```diff
@@ -91,14 +88,18 @@ def __init__(self, minidump_file, *, trace=False, quiet=False, thread_id=None):
         self._allocate_size = 1024 * 1024 * 10  # NOTE: 10 megs
         self._allocate_ptr = None
         self._setup_emulator(thread)
+        # hacky but ¯\_(ツ)_/¯
+        from .modules import ModuleManager, Module

```

Comment:
I think this is actually a pretty big (fundamental) problem. You are creating a circular dependency on the `Dumpulator` class which means the `ModuleManager` likely has too much responsibility. Perhaps it will be easier if the `ModuleManager` just has the following responsibilities:

- Keep a (sorted) `List[Module]` so you can do `dp.modules.find(0x12345)` and it finds the matching `Module`
- The `Module` can have a constructor that takes the `pefile.PE` and extracts the exports
- `dp.modules.find("ntdll.dll") -> Module`

Probably the whole `Peb` thing can be left out of scope (for now), pretty sure it's almost redundant with the new `HandleManager` anyway.

Reply:
Oh and to be fair, https://github.com/mrexodia/dumpulator/issues/17 was totally under-specified. It was more of a reminder for myself and totally not thought out, so sorry for wasting so much time...

Reply:
Yeah I understand, let me close this PR and rework it. 

I do like the functionality of the current implementation I just need to rework it to remove the `Dumpulator` dependency.

I'll go through and move around the code so if some functionality completely relies on `Dumpulator` I will put it into the main class.